collect_app/src/androidTest/java/org/odk/collect/android/feature/projects/GoogleDriveDeprecationBannerTest.kt

```diff
@@ -0,0 +1,125 @@
+            .openGooglePickerAndSelect(googleAccount)
+            .assertText(R.string.google_drive_deprecation_message)
+    }
+
+    @Test
+    fun forumThreadIsOpenedAfterClickingLearnMore() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .clickOnString(R.string.learn_more_button_text)
+
+        intended(
+            allOf(
+                hasComponent(WebViewActivity::class.java.name),
+                hasExtra("url", "https://forum.getodk.org/t/40097")
+            )
+        )
+    }
+
+    @Test
+    fun dismissButtonIsVisibleOnlyAfterClickingLearnMore() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .assertTextNotDisplayed(R.string.dismiss_button_text)
+            .clickOnString(R.string.learn_more_button_text)
+            .pressBack(MainMenuPage())
+            .assertText(R.string.dismiss_button_text)
+    }
+
+    @Test
+    fun afterClickingDismissTheBannerDisappears() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .clickOnString(R.string.learn_more_button_text)
+            .pressBack(MainMenuPage())
+            .clickOnString(R.string.dismiss_button_text)
+            .assertTextNotDisplayed(R.string.google_drive_deprecation_message)
+            .rotateToLandscape(MainMenuPage())
+            .assertTextNotDisplayed(R.string.google_drive_deprecation_message)
+    }
+
+    @Test
+    fun dismissingTheBannerInOneProjectDoesNotAffectOtherProjects() {
+        val googleAccount = "steph@curry.basket"
+        testDependencies.googleAccountPicker.setDeviceAccount(googleAccount)
+
+        rule.startAtMainMenu()
+            .openProjectSettingsDialog()
+            .clickAddProject()
+            .switchToManualMode()
+            .openGooglePickerAndSelect(googleAccount)
+            .clickOnString(R.string.learn_more_button_text)
+            .pressBack(MainMenuPage())
+            .clickOnString(R.string.dismiss_button_text)
+            .assertTextNotDisplayed(R.string.google_drive_deprecation_message)
+            .openProjectSettingsDialog()
+            .clickSettings()
+            .clickProjectManagement()
+            .deleteProject()
```

Comment:
I don't think you need to delete the first project in this test. In fact, deleting might actually allow a bad implementation where the banner is hidden for all projects if it's dismissed in one (which wouldn't be caught in the test if the project is deleted).

Reply:
Ok, I've updated the test so now it adds a new GD project. Doing that I discovered a bug... If you add two GD projects the second one is treated as a duplicate and there is a dialog that asks if you want to add it or switch to the existing one. That's why I've updated this method: https://github.com/getodk/collect/pull/5414/commits/c15d4ca24e3b925965fd35b2b34d7434fb5679f4#diff-37878b479841035eb4ab6e19b89c4057744c0f7a948dcebebff40f9607cb39c2R36
I didn't file it because I think we can ignore it:
- I don't think people use two different GD projects
- no one has complained about it
- we are going to remove the GD integration